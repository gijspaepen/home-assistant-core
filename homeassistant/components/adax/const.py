"""Constants for the Adax integration."""

from typing import Final

from homeassistant.const import Platform

ACCOUNT_ID: Final = "account_id"
CLOUD = "Cloud"
CONNECTION_TYPE = "connection_type"
DOMAIN: Final = "adax"
LOCAL = "Local"
WIFI_SSID = "wifi_ssid"
WIFI_PSWD = "wifi_pswd"

PLATFORMS = [Platform.CLIMATE, Platform.SENSOR]
