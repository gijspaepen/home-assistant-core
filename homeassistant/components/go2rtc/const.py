"""Go2rtc constants."""

DOMAIN = "go2rtc"

CONF_DEBUG_UI = "debug_ui"
EXCLUSE_ERROR_MESSAGE = "Url and debug_ui cannot be set at the same time."
