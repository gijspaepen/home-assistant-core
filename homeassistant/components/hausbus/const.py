"""Constants for the Haus-Bus integration."""

from __future__ import annotations

import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "hausbus"
ATTR_ON_STATE = "on_state"
