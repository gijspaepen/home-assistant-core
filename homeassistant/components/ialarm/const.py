"""Constants for the iAlarm integration."""

from pyialarm import IAlarm

from homeassistant.components.alarm_control_panel import AlarmControlPanelEntityState

DATA_COORDINATOR = "ialarm"

DEFAULT_PORT = 18034

DOMAIN = "ialarm"

IALARM_TO_HASS = {
    IAlarm.ARMED_AWAY: AlarmControlPanelEntityState.ARMED_AWAY,
    IAlarm.ARMED_STAY: AlarmControlPanelEntityState.ARMED_HOME,
    IAlarm.DISARMED: AlarmControlPanelEntityState.DISARMED,
    IAlarm.TRIGGERED: AlarmControlPanelEntityState.TRIGGERED,
}
