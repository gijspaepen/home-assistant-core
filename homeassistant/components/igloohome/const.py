"""Constants for the igloohome integration."""

DOMAIN = "igloohome"

OAUTH2_AUTHORIZE = "https://auth.igloohome.co/login"
OAUTH2_TOKEN = "https://auth.igloohome.co/oauth2/token"
