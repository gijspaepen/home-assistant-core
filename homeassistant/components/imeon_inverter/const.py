"""Constant for Imeon component."""

DOMAIN = "imeon_inverter"
HUBNAME = "imeon_inverter_hub"
TIMEOUT = 10  # seconds
