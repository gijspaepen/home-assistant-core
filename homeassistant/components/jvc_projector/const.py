"""Constants for the jvc_projector integration."""

from jvcprojector import const

NAME = "JVC Projector"
DOMAIN = "jvc_projector"
MANUFACTURER = "JVC"

REMOTE_COMMANDS = {
    "menu": const.REMOTE_MENU,
    "up": const.REMOTE_UP,
    "down": const.REMOTE_DOWN,
    "left": const.REMOTE_LEFT,
    "right": const.REMOTE_RIGHT,
    "ok": const.REMOTE_OK,
    "back": const.REMOTE_BACK,
    "mpc": const.REMOTE_MPC,
    "hide": const.REMOTE_HIDE,
    "info": const.REMOTE_INFO,
    "input": const.REMOTE_INPUT,
    "cmd": const.REMOTE_CMD,
    "advanced_menu": const.REMOTE_ADVANCED_MENU,
    "picture_mode": const.REMOTE_PICTURE_MODE,
    "color_profile": const.REMOTE_COLOR_PROFILE,
    "lens_control": const.REMOTE_LENS_CONTROL,
    "setting_memory": const.REMOTE_SETTING_MEMORY,
    "gamma_settings": const.REMOTE_GAMMA_SETTINGS,
    "hdmi_1": const.REMOTE_HDMI_1,
    "hdmi_2": const.REMOTE_HDMI_2,
    "mode_1": const.REMOTE_MODE_1,
    "mode_2": const.REMOTE_MODE_2,
    "mode_3": const.REMOTE_MODE_3,
    "lens_ap": const.REMOTE_LENS_AP,
    "gamma": const.REMOTE_GAMMA,
    "color_temp": const.REMOTE_COLOR_TEMP,
    "natural": const.REMOTE_NATURAL,
    "cinema": const.REMOTE_CINEMA,
    "anamo": const.REMOTE_ANAMO,
    "3d_format": const.REMOTE_3D_FORMAT,
}
