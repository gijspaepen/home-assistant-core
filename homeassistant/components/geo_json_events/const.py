"""Define constants for the GeoJSON events integration."""

from __future__ import annotations

from typing import Final

from homeassistant.const import Platform

DOMAIN: Final = "geo_json_events"

PLATFORMS: Final = [Platform.GEO_LOCATION]

ATTR_EXTERNAL_ID: Final = "external_id"
DEFAULT_RADIUS_IN_KM: Final = 20.0
DEFAULT_RADIUS_IN_M: Final = 20000.0
DEFAULT_UPDATE_INTERVAL: Final = 300
SOURCE: Final = "geo_json_events"
CONF_UPDATE_INTERVAL: Final = "update_interval"

SIGNAL_DELETE_ENTITY: Final = "geo_json_events_delete_{}"
SIGNAL_UPDATE_ENTITY: Final = "geo_json_events_update_{}"
