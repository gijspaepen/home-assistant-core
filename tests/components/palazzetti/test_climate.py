"""Tests for the Palazzetti climate platform."""

from unittest.mock import AsyncMock, patch

from homeassistant.config_entries import ConfigEntryState
from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from tests.common import MockConfigEntry

CLIMATE_ID = f"{Platform.CLIMATE}.stove"


async def test_climate(
    hass: HomeAssistant,
    entity_registry: er.EntityRegistry,
    mock_config_entry: MockConfigEntry,
    mock_palazzetti_client: AsyncMock,
) -> None:
    """Test the creation and values of Palazzetti climate device."""
    patch("pypalazzetti.client.PalazzettiClient", mock_palazzetti_client)

    mock_config_entry.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    assert mock_config_entry.state is ConfigEntryState.LOADED
    assert entity_registry.async_is_registered(CLIMATE_ID)

    entity = entity_registry.async_get(CLIMATE_ID)

    assert entity.unique_id == "11:22:33:44:55:66"
