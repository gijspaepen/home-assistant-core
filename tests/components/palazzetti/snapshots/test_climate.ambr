# serializer version: 1
# name: test_all_entities[example1][climate.stove-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'fan_modes': list([
        '1',
        '2',
        '3',
        '4',
        '5',
        'HIGH',
        'AUTO',
      ]),
      'hvac_modes': list([
        <HVACMode.HEAT: 'heat'>,
        <HVACMode.OFF: 'off'>,
      ]),
      'max_temp': 50,
      'min_temp': 5,
      'target_temp_step': 1.0,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'climate',
    'entity_category': None,
    'entity_id': 'climate.stove',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': None,
    'platform': 'palazzetti',
    'previous_unique_id': None,
    'supported_features': <ClimateEntityFeature: 393>,
    'translation_key': None,
    'unique_id': '11:22:33:44:55:66',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[example1][climate.stove-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'current_temperature': 18,
      'fan_mode': '3',
      'fan_modes': list([
        '1',
        '2',
        '3',
        '4',
        '5',
        'HIGH',
        'AUTO',
      ]),
      'friendly_name': 'Stove',
      'hvac_modes': list([
        <HVACMode.HEAT: 'heat'>,
        <HVACMode.OFF: 'off'>,
      ]),
      'max_temp': 50,
      'min_temp': 5,
      'supported_features': <ClimateEntityFeature: 393>,
      'target_temp_step': 1.0,
      'temperature': 21,
    }),
    'context': <ANY>,
    'entity_id': 'climate.stove',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'heat',
  })
# ---
# name: test_all_entities[example2][climate.stove2-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'fan_modes': list([
        'SILENT',
        '1',
        '2',
        '3',
        '4',
        '5',
        'AUTO',
      ]),
      'hvac_modes': list([
        <HVACMode.HEAT: 'heat'>,
        <HVACMode.OFF: 'off'>,
      ]),
      'max_temp': 50,
      'min_temp': 5,
      'target_temp_step': 1.0,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'climate',
    'entity_category': None,
    'entity_id': 'climate.stove2',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': None,
    'platform': 'palazzetti',
    'previous_unique_id': None,
    'supported_features': <ClimateEntityFeature: 393>,
    'translation_key': None,
    'unique_id': '11:22:33:44:55:66',
    'unit_of_measurement': None,
  })
# ---
# name: test_all_entities[example2][climate.stove2-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'current_temperature': 22,
      'fan_mode': '5',
      'fan_modes': list([
        'SILENT',
        '1',
        '2',
        '3',
        '4',
        '5',
        'AUTO',
      ]),
      'friendly_name': 'Stove2',
      'hvac_modes': list([
        <HVACMode.HEAT: 'heat'>,
        <HVACMode.OFF: 'off'>,
      ]),
      'max_temp': 50,
      'min_temp': 5,
      'supported_features': <ClimateEntityFeature: 393>,
      'target_temp_step': 1.0,
      'temperature': 21,
    }),
    'context': <ANY>,
    'entity_id': 'climate.stove2',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'heat',
  })
# ---
